dso-l2/src/main/java/com/tc/objectserver/managedobject/ConcurrentDistributedServerMapManagedObjectState.java

```diff
@@ -588,19 +588,23 @@ private void applyClearVersioned(ApplyTransactionInfo applyInfo) {
   }
 
   private void applyRegisterServerEventListener(ApplyTransactionInfo applyInfo, Object[] params) {
-    ClientID clientID = (ClientID) applyInfo.getServerTransactionID().getSourceID();
+    ClientID clientID = params[0] instanceof Long ? new ClientID((Long)params[0]) : (ClientID)applyInfo.getServerTransactionID().getSourceID();
     for (Object eventTypeIndex : params) {
-      ServerEventType serverEventType = ServerEventType.values()[(Integer) eventTypeIndex];
-      eventRegistry.put(serverEventType, clientID);
-      applyInfo.getClientChannelMonitor().monitorClient(clientID, getId());
+      if (eventTypeIndex instanceof Integer) {
```

Comment:
Once the parameter order is fixed on the client side you shouldn't need to watch out for non Integer instances on this side.